src/transport-http-client.ts

```diff
@@ -58,28 +63,50 @@ export class TransportHttpClient<BagType extends FnsBag> implements ITransport<B
                 if (conn.isClosed) throw new RpcErrorUseAfterClose('the connection is closed');
                 log(`connection "${conn.description}" is sending an envelope:`, env);
                 log('send...');
+
+                conn.status.set('CONNECTING');
+                const urlToPost = url + `from/${this.deviceId}`;
+                log(`send... POSTing to ${urlToPost}`);
+
+                const { request, cancel: cancelRequest } = fetchWithTimeout(
+                    TIMEOUT,
+                    urlToPost,
+                    {
+                        method: 'POST',
+                        headers: {
+                            'Accept': 'application/json',
+                            'Content-Type': 'application/json',
+                        },
+                        body: JSON.stringify([env]),
+                    },
+                );
+
+                conn.onClose(() => {
+                    cancelRequest();
+                });
+
                 try {
-                    conn.status.set('CONNECTING');
-                    const urlToPost = url + `from/${this.deviceId}`;
-                    log(`send... POSTing to ${urlToPost}`);
-                    const res = await withTimeout(
-                        TIMEOUT,
-                        fetch(urlToPost, {
-                            method: 'POST',
-                            headers: {
-                                'Accept': 'application/json',
-                                'Content-Type': 'application/json',
-                            },
-                            body: JSON.stringify([env]), // a batch of one envelope
-                        }),
-                    );
+                    const res = await request;
+
+                    if (this.isClosed) return;
+
                     if (!res.ok) {
                         log('send... POST was not ok...');
                         throw new RpcErrorNetworkProblem(
                             `a POST to ${urlToPost} resulted in http ${res.status}`,
                         );
                     } else {
                         log('send... success.');
+
+                        // cancel current pull timers for this connection
```

Comment:
Cancelling HTTP requests can be dangerous here - if we cancel a connection while it's transferring a large Envelope, I think that Envelope will be lost in the void since the server will believe it's sent.

Reply:
Fortunately this is not what happens! Rescheduling does not cancel the previous fetch, it just makes it so that the next poll comes sooner. The only time a fetch request is actually cancelled is when the connection is closed.